lib/puppet/network/http/pool.rb

```diff
@@ -9,11 +9,9 @@
 # @api private
 #
 class Puppet::Network::HTTP::Pool < Puppet::Network::HTTP::BasePool
-  FIFTEEN_SECONDS = 15
+  attr_reader :factory, :keepalive_timeout
 
-  attr_reader :factory
-
-  def initialize(keepalive_timeout = FIFTEEN_SECONDS)
+  def initialize(keepalive_timeout)
```

Comment:
I'm not sure I follow the logic behind requiring a timeout here. Is there a reason we don't just default to `Puppet[:http_keepalive_timeout]` rather than `FIFTEEN_SECONDS`?

Reply:
I've been trying to make an effort to resolve puppet settings at the application/subsystem level and pass state explicitly. That way we're not relying on global state, which may be mutable(!). However, I haven't been 100% consistent, as some of the services still reference `Puppet[:server]`, etc.

Reply:
Ah, okay, that does make sense. I didn't think about it from that perspective. Cool, thanks!